### PR TITLE
#166789868 User sign-up and authentication

### PR DESCRIPTION
**Pull request by adafia**
#### What does this PR do?
This pull request:
 - update user tests to test for token generation
 - hash user password
 - authenticate user
 - save user data to the database
 - add unit test for user login
 - add user login endpoint
 - document login endpoint
 - add build script to package.json

#### Description of Task to be completed?
 - The task was to have a token generated any time a user signs up or logs in to the application.
 - All passwords inputted by the user while signing up are also hashed

>Then do the following steps to manually test on your local machine
 - run ``` npm run start ```
 - access the user sign-up route via ```/api/users``` on the HTTP ```POST``` method.
 - access the user login route via ```/api/users/login``` on the HTTP ```POST``` method.
 - inspect the body of the response for the presence of a token.

>To manually test remotely, do the following: 
 - access the base rout via this [link](ah-lobos-backend-swagger.herokuapp.com)
 - access the user sign-up route by adding ```/api/users``` to the base route.
 - access the user login route by adding ```/api/users/login``` to the base route.
 - inspect the body of the response for the presence of a token.

#### Any background context you want to provide?
The JWTs are required to authenticate a user's access to the various resources of the application.

#### What are the relevant pivotal tracker stories?
[#166789868](https://www.pivotaltracker.com/story/show/166789868)

#### Screenshots (if appropriate)
Screen Shot 2019-07-02 at 16 01 45

#### Questions: 
N/A

### CODE REVIEW

**Review by manzif — APPROVED** [review]

File: src/middlewares/auth.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import dotenv from 'dotenv';
+import bcrypt from 'bcrypt';
+
+dotenv.config();
+
+class Authorization {
+  static async genToken(req, res, next) {
+    const userData = req.body;
+    userData.password = bcrypt.hashSync(userData.password, 10);
+    const accesstoken = await jwt.sign(userData, process.env.SECRET_JWT_KEY);
```
Comment (Dom58) [inline src/middlewares/auth.js]: You have also to add the period that the generated token will be expired

File: src/controllers/userControllers.js [hunk 2]
```diff
@@ -14,17 +16,26 @@ class UserManager {
    */
   static async registerUser(req, res) {
     try {
-      const { username, email, password } = req.body;
+      const {
+        username, email, password, bio, image
```
Comment (j4l13n) [inline src/controllers/userControllers.js:20]: Why adding bio and image when signing up?
I think you should remove bio and image

**Review by j4l13n — APPROVED** [review]

**Review by Emile-Nsengimana — APPROVED** [review]

**Review by Dom58 — APPROVED** [review]

File: src/db/config/config.js [hunk 3]
```diff
@@ -11,11 +11,11 @@ const config = {
...
   test: {
     username: envConfig.db_username_test,
-    password: envConfig.db_password,
+    password: envConfig.db_password_test,
     database: 'ah_db_test',
-    host: '127.0.0.1',
+    host: 'localhost',
     dialect: 'postgres',
-    operatorsAliases: false
+    // operatorsAliases: false
```
Comment (emukungu) [inline src/db/config/config.js]: remove the commented out code

**Review by j4l13n — APPROVED** [review]

**Comment by emukungu** [comment]
@adafia, reduce the commit body to only 3 points

**Pull request merged**